File: src/jiji/rpc/converters/account_converter.rb
# coding: utf-8

require 'grpc'

module Jiji::Rpc::Converters
  module AccountConverter
    include Jiji::Rpc

    def convert_account_to_pb(account)
      return nil unless account
      hash = account.to_h
      convert_numeric_to_pb_decimal(hash, [
        :balance,
        :profit_or_loss,
        :margin_used,
        :margin_rate
      ])
      Account.new(convert_hash_values_to_pb(hash))
    end
  end
end

File: agent_services/python/src/jiji/rpc/converters/account_converter.py
from jiji.model.account import Account
from jiji.rpc.converters.primitive_converter import convert_decimal_from, convert_timestamp_from

def convert_account_from(pb_account):
    if (pb_account == None):
        return None
    return Account(pb_account.account_id, pb_account.account_currency, \
        convert_decimal_from(pb_account.balance), \
        convert_decimal_from(pb_account.profit_or_loss), \
        convert_decimal_from(pb_account.margin_used), \
        convert_decimal_from(pb_account.margin_rate), \
        convert_timestamp_from(pb_account.updated_at))
